src/diffusers/pipelines/unidiffuser/pipeline_unidiffuser.py

```diff
@@ -0,0 +1,1394 @@
+import inspect
+from dataclasses import dataclass
+from typing import Callable, List, Optional, Union
+
+import einops
```

Comment:
We don't have a dependency on `einops` can we remove this one here please? :-)

Reply:
We will need to refactor all those einops operations I'm afraid :-/

Reply:
Should have removed the dependency on `einops` by refactoring all `einops` operations to pure `torch` operations.